src/main/java/calculator/domain/StringCalculator.java

```diff
@@ -0,0 +1,56 @@
+package calculator.domain;
+
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringCalculator {
+
+    private static final String CUSTOM_DELIMITER_PATTERN = "//(.)\n(.*)";
+    private String delimiter = ",|:";
+
+
+    public StringCalculator() {
+    }
+
+    public int add(String inputString) {
+        if(StringUtils.isEmpty(inputString)) return 0;
+
+        String[] tokens = getTokens(inputString);
+
+        return Arrays.stream(tokens)
+                .peek(token -> {
+                    if (isNotNumber(token) || isNegativeNumber(token)) throw new RuntimeException("음수 또는 문자열은 계산할 수 없습니다.");
+                })
+                .collect(Collectors.summingInt(Integer::parseInt));
+    }
+
+    private String[] getTokens(String inputString) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_PATTERN).matcher(inputString);
+        if (m.find()) {
+            addDelimeter(m.group(1));
+            return m.group(2).split(delimiter);
+        }
+        return inputString.split(delimiter);
+    }
+
+    private void addDelimeter(String substring) {
+        this.delimiter += "|" + substring;
+    }
+
+    private boolean isNegativeNumber(String number) {
+        return Integer.parseInt(number) < 0 ? true : false;
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return Integer.parseInt(number) < 0;
```